x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/api/create.ts

```diff
@@ -54,7 +53,17 @@ export const registerSiemRuleMigrationsCreateRoute = (
             const ctx = await context.resolve(['securitySolution']);
             const ruleMigrationsClient = ctx.securitySolution.getSiemRuleMigrationsClient();
 
-            await siemMigrationAuditLogger.logCreateMigration({ migrationId });
+            let migrationId: string;
+
+            if (!providedMigrationId) {
+              /** if new migration */
+              migrationId = await ruleMigrationsClient.data.migrations.create();
+              await siemMigrationAuditLogger.logCreateMigration({ migrationId });
+            } else {
+              /** if updating existing migration */
+              migrationId = providedMigrationId;
+              await siemMigrationAuditLogger.logUpdateMigration({ migrationId });
```

Comment:
Not sure about the "update migration" action. I would reserve that name for actions like updating the migration name. Here, the action is more "upload migration rules", right?

Reply:
Firstly, imo i think sending a request to `create` endpoint with a populated `migrationId` is a little misleading because it is actually not creating a migration as `endpoint` suggests, rather updating it ( logically ) by adding more rules to it.

That is a thought process behind adding this log. Alternatively, we can also add a separate function called `addRulesToMigration` to be more precise.

In all, i think we should do a log when additional rules are being added to a `migration`.

Reply:
Yes, I agree, this `create` route was reused for the 2 scenarios (with and without ID) because the logic was almost the same; the difference was only one line.

Now it seems the logic is diverging more because we need to create the migration document. So, I would be okay splitting this into 2 different routes:

- Create [`POST /siem_migrations/rules`] - Creates the migration document (without rules) and returns the migration_id. It receives the settings in the body.
- Upload [`POST /siem_migrations/rules/{migration_id}`] - Creates the rules for a specific migration

Each route will have their own audit logs as well.

wdyt?

Reply:
I agree with the idea of creating separate routes and we can split this one as and when needed i.e. when we add settings in near future.

I will do a slight modification to your suggestion.

- `POST /seim_migrations/` - creates new migration ( could be of type `rule` of `dashboard` )
- `POST /siem_migrations/{migration_id}` - updates an existing migration. Granted we do not have this use case of now but assuming it for the sake of future proofing.
- `POST /siem_migrations/rules/{migration_id}` - Add rules to the migration batch.

Reply:
Closing this comment since we will be discussing this separate call.

Reply:
Removed `logUpdateMigration` in https://github.com/elastic/kibana/pull/216164/commits/43b91c5fdbe23f1f1a6f98b935c9e6ccd12f928d